core/web/evm_transfer_controller.go

```diff
@@ -104,11 +105,14 @@ func ValidateEthBalanceForTransfer(c *gin.Context, chain evm.Chain, fromAddr com
 		return errors.Wrap(err, "failed to estimate gas")
 	}
 
-	// TODO: support EIP-1559 transactions
-	if fees.Legacy == nil {
-		return errors.New("estimator did not return legacy tx fee estimates")
+	var gasPrice *assets.Wei
```

Comment:
Not a blocker - but is this conditional based on `fees.Legacy` the right level of implementation detail for this method? It feels like `ValidateEthBalanceForTransfer` should be responsible for getting the gas price, but that the actual calculation (regardless of legacy vs not) should happen in the layer below. 

`GetFee` is defined on the estimator - can we handle the legacy vs non-legacy gas calculation there? The receiver of `GetFee` (`WrappedEvmEstimator`) has access to `EIP1559Enabled` which should be fairly clean.

Reply:
just checking my understanding - are you suggesting that that the `GetFee` func (or something like it) should just return the total ETH rather than the gas price?

Reply:
L103 - 120 all build up to calculating `amountWithFees` and include calculations that are implicitly dependent on the value of wrappedEvmEstimator.EIP1559Enabled. If that could live as an API defined on `WrappedEvmEstimator` I think it would expose the right level of implementation details to the caller and be more readable. 

Code sample: https://github.com/smartcontractkit/chainlink/pull/10208

Reply:
makes sense! it's a cleaner separation where any fee calculation is handed off to the estimator

I will take a look at your draft PR :)